batteries/result.luau

```diff
@@ -0,0 +1,31 @@
+type T<Value> = {
+	success: true,
+	value: Value,
+} | {
+	success: false,
+	traceback: string,
+	err: string,
```

Comment:
Do we want this over a form where `err` can be specified?

Reply:
Yes. The reason why is so that result failures can be appended. For example:
```luau
local function someOperation()
  return result(false, "failed to do generic operation")
end

local function meow()
  local call = someOperation()
  if not call then
    call.err = `failure in function "meow"; {call.err}`
  end
end

local function meow2()
  local call = someOperation()
  if not call then
    call.err = `failure in function "meow2"; {call.err}`
  end
end
```
err as string is *great* for propagating messages. Maybe we could attach something like `errMessage` and `err` to have both? I can see the value in, say, an error tagged enum.